rust/src/modbus/detect.rs

```diff
@@ -0,0 +1,563 @@
+/* Copyright (C) 2021 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+use super::modbus::ModbusTransaction;
+use lazy_static::lazy_static;
+use regex::Regex;
+use sawp_modbus::{AccessType, CodeCategory, Data, Flags, FunctionCode, Message};
+use std::ffi::CStr;
+use std::ops::{Range, RangeInclusive};
+use std::os::raw::{c_char, c_void};
+use std::str::FromStr;
+
+lazy_static! {
```

Comment:
I wonder if regular expressions are the best way to go here...
What do you think ?
see also https://redmine.openinfosecfoundation.org/issues/4112

Reply:
I think these are two separate discussions. Both were discussed previously in #5790.

**Use of regexes versus nom to parse arguments**
https://github.com/OISF/suricata/pull/5790#discussion_r564789021

**Use of Range Vs. DetectU32Data**
https://github.com/OISF/suricata/pull/5790#discussion_r564790700

My opinion for both would be to leave as-is for now in the interest of merging this PR sooner. Create a generic module for `DetectU32Data` along with a nom parser for arguments as stated in [4112](https://redmine.openinfosecfoundation.org/issues/4112). Once it is available, refactor modbus to use it.

Reply:
Thanks for bringing back the pointers. I had a feeling of déjà-vu when saying this.
Fair enough for me